docs/docs/creating-a-generic-plugin.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Creating a Generic plugin
+---
+
+This section is aimed at explaining the structure of a Gatsby plugin and the files that are needed to create it.
+
+As seen in the [what is a plugin doc](/docs/what-is-a-plugin/),a plugin is a piece of software that acts as an add-on and gives your site additional functionality.
+
+They contain a file, usually in the project root, called `package.json` - this file holds various metadata relevant to the project. This file is used to give information to npm that allows it to identify the project as well as handle the project's dependencies.
+
+The process of initializing a `package.json` in a project is done by running :
```

Comment:
There should probably be a heading here. Maybe ## Initializing your plugin project

Reply:
I agree.